### PR TITLE
OOP Exercise (update tiny JS world)

### PR DESCRIPTION
**Pull request by andreymashko7**
### OOP Exercise

[Demo](https://andreymashko7.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/andreymashko7/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by andreymashko7** [comment]
self-check is done

File: submissions/AndriyM/oop_exercise_tiny_world/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+class Inhabitant {
+    constructor(species, name, gender, legs, saying) {
+        this.species = species;
+        this.name = name;
+        this.legs = legs;
+        this.gender = gender;
+        this.saying = saying;
+    }
+    inhabitantShowProps() {
+        return `${this.species}; ${this.name}; ${this.gender}; ${this.legs}; ${this.saying}`;
```
Comment (OleksiyRudenko) [inline submissions/AndriyM/oop_exercise_tiny_world/index.js:10]: Make this fragment DRY

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@andreymashko7 well done.
Let's polish it to see that you embraced DRY ideology.

**Comment by andreymashko7** [comment]
Please, review again

File: submissions/AndriyM/oop_exercise_tiny_world/index.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class Inhabitant {
+	constructor(species, name, gender, legs, saying) {
+		this.species = species;
+		this.name = name;
+		this.legs = legs;
+		this.gender = gender;
+		this.saying = saying;
+	}
+	inhabitantShowProps() {
+		return keys.map((key) => this[key]).join("; ");
```
Comment (OleksiyRudenko) [inline submissions/AndriyM/oop_exercise_tiny_world/index.js:11]: `keys` and `key` are generic terms.
Any good reason to keep list of properties outside class definition?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@andreymashko7 quite a decent job.
Let's polish it.

**Comment by andreymashko7** [comment]
Better now? I hope I got it right.

**Comment by OleksiyRudenko** [comment]
@andreymashko7 only second half of mentor's comment is addressed.

**Comment by andreymashko7** [comment]
Fixed,please check

**Comment by andreymashko7** [comment]
@OleksiyRudenko , please check

**Review by OleksiyRudenko — APPROVED** [review]
@andreymashko7 great job!

**Pull request merged**